library/Fission/AWS/Route53.hs

```diff
@@ -0,0 +1,77 @@
+module Fission.AWS.Route53 
+  ( createChangeRequest
+  , registerDomain
+  ) where
+
+import RIO
+
+import Data.Has
+import qualified Fission.Config as Config
+
+import Network.AWS.Route53 as Route53
+import Network.AWS.Prelude as AWS
+import Network.AWS.Auth    as AWS
+import Fission.AWS.Types   as AWS
+import Control.Monad.Trans.AWS
+
+import Fission.Internal.Constraint
+import Fission.IPFS.CID.Types
+
+
+import Control.Lens ((?~))
+
+registerDomain :: MonadRIO      cfg m
+           => Has AWS.AccessKey cfg
+           => Has AWS.SecretKey cfg
+           => Has AWS.ZoneId    cfg
+           => Has AWS.Domain    cfg
+           => Text
+           -> CID
+           -> m(AWS.Domain)
+registerDomain subdomain cid= do 
+  Domain domain <- Config.get
+  env <- createEnv
+  let
+    baseUrl    = subdomain <> domain
+    dnslinkUrl = "_dnslink" <> baseUrl
+    dnslink    = "dnslink=/ipfs/" <> (unaddress cid)
+
+  baseReq    <- createChangeRequest Cname baseUrl "ipfs.runfission.com"
+  dnslinkReq <- createChangeRequest Txt dnslinkUrl $ wrapQuotes dnslink
+
+  liftIO $ runResourceT . runAWST env . within NorthVirginia $ do
+    void $ send(baseReq)
+    void $ send(dnslinkReq)
+    return $ AWS.Domain baseUrl
```

Comment:
I know that this is a WIP, but I feel like we could break something like this into a helper:

```haskell
-- Helper --

runAWS = liftIO . runResourceT . runAWST

-- Use --

runAWS env $ within NorthVirginia do
  send baseReq
  send dnsLinkReq
  return $ AWS.Domain baseURL
```

Reply:
For context, we use `runResourceT` in the DB, too, but you never have to deal with it directly because it's tucked away in a helper.

Reply:
I'm having an awful time with the types on this one... May need to run it by you at some point

Reply:
@dholms sure thing!